### PR TITLE
ref(ui): Convert indicators container from createReactClass

### PR DESCRIPTION
**Pull request by scttcper**

### CODE REVIEW

**Review by billyvg — APPROVED** [review]
nice

File: static/app/components/indicators.tsx [hunk 1]
```diff
@@ -54,15 +52,19 @@ class Indicators extends React.Component<Props> {
...
+  }
 
-  getInitialState() {
-    return {
-      items: [],
-    };
-  },
+  unlistener = IndicatorStore.listen((items: Indicator[]) => {
+    this.setState({items});
+  }, undefined);
```
Comment (EvanPurkhiser) [inline static/app/components/indicators.tsx:67]: oh this is nice, we should definitely change all of our reflux usage to this type of pattern and kill the mixins if we can.

File: static/app/components/indicators.tsx [hunk 2]
```diff
@@ -54,15 +52,19 @@ class Indicators extends React.Component<Props> {
...
+  componentWillUnmount() {
+    this.unlistener?.();
+  }
 
-  getInitialState() {
-    return {
-      items: [],
-    };
-  },
+  unlistener = IndicatorStore.listen((items: Indicator[]) => {
```
Comment (EvanPurkhiser) [inline static/app/components/indicators.tsx:65]: I like to call these `disposer`

Comment (billyvg) [reply static/app/components/indicators.tsx:65]: We already use `unlistener` in a few other places fwiw

**Review by markstory — APPROVED** [review]

**Pull request merged**